OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/PermissionsForRoleControllerTests.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using NUnit.Framework;
+using OutOfSchool.Common.PermissionsModule;
+using OutOfSchool.Services.Enums;
+using OutOfSchool.WebApi.Controllers.V1;
+using OutOfSchool.WebApi.Models;
+using OutOfSchool.WebApi.Services;
+
+namespace OutOfSchool.WebApi.Tests.Controllers
+{
+    [TestFixture]
+    public class PermissionsForRoleControllerTests
+    {
+        private PermissionsForRoleController controller;
+        private Mock<IPermissionsForRoleService> service;
+
+        private IEnumerable<PermissionsForRoleDTO> permissionsForAllRoles;
+        private PermissionsForRoleDTO permissionsForRoleDTO;
+
+
+
+        [SetUp]
+        public void Setup()
+        {
+            service = new Mock<IPermissionsForRoleService>();
+
+            controller = new PermissionsForRoleController(service.Object);
+
+            permissionsForAllRoles = FakePermissionsForAllRoles();
+            permissionsForRoleDTO = FakePermissionsForRole();
+        }
+
+        [Test]
+        public async Task GetsAllPermissionsForRoles_WhenCalled_ReturnsOkObjectResult()
+        {
+            // Arrange
+            service.Setup(x => x.GetAll()).ReturnsAsync(permissionsForAllRoles);
+
+            // Act
+            var response = await controller.GetAll().ConfigureAwait(false) as ObjectResult;
+
+            // Assert
+            Assert.IsInstanceOf<OkObjectResult>(response);
```

Comment:
My guess is that `Assert.IsInstanceOf<OkObjectResult>(response);` check the type of response and if it is different it states in error message that expected was `OkObjectResult` but actual is `X`.
This has less sense as you've already lost type of `await controller.GetAll().ConfigureAwait(false)` and `X` will always be `ObjectResult` (because of casting `as ObjectResult`)
Is it possible to perform casting later? (here and in all other tests)